src/lib.rs

```diff
@@ -464,44 +466,17 @@ pub enum Edition {
     /// Edition 2021
     #[serde(rename = "2021")]
     E2021,
-    /// An unknown value was passed instead of a valid edition.
-    ///
-    /// This can mean one of two things. Either some corrupted data, or an edition not supported by the current crate version.
-    /// If you believe the latter is the case feel free to open an issue.
-    Unknown(Cow<'static, str>),
-}
-
-impl Hash for Edition {
-    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-        match self {
-            Edition::Unknown(ref unknown) => match unknown.as_ref() {
-                "2015" => Self::E2015.hash(state),
-                "2018" => Self::E2018.hash(state),
-                "2021" => Self::E2021.hash(state),
-                _ => {
-                    core::mem::discriminant(unknown).hash(state);
-                    core::mem::discriminant(self).hash(state);
-                }
-            },
-            _ => core::mem::discriminant(self).hash(state),
-        }
-    }
-}
-
-impl PartialEq for Edition {
-    fn eq(&self, other: &Self) -> bool {
-        match (self, other) {
-            (Self::Unknown(left), Self::Unknown(right)) => left == right,
-            (Self::E2015, Self::Unknown(unknown)) => unknown == "2015",
-            (Self::E2018, Self::Unknown(unknown)) => unknown == "2018",
-            (Self::E2021, Self::Unknown(unknown)) => unknown == "2021",
-            _ => core::mem::discriminant(self) == core::mem::discriminant(other),
-        }
-    }
+    /// Edition 2024
+    #[serde(rename = "2024")]
```

Comment:
Maybe hide the from the docs for now, it's not like matching on them explicitly helps anyone at present

Reply:
I agree it probably doesn't make any sense to show them, I'll also mark it as non_exhaustive to make it less confusing

Reply:
Maybe it's a good idea to also prefix them with an underscore for now?

Reply:
hmmm... because we only need to parse them successfully, not make them accessible?

Reply:
Yeah exactly, the edition 2024 and so forth don't exist yet so relying on that isn't right. We want to parse them so that old code will still parse correctly when a new edition comes out but it can't possibly make sense of it since it didn't exist at the time. When 2024 actually comes out we can remove the prefix unhide it from the docs